test/aleph/http_test.clj

```diff
@@ -1656,6 +1658,55 @@
         (is (= true (deref conn-closed 1000 :timeout)))
         (is (thrown? RequestCancellationException (deref rsp 1000 :timeout)))))))
 
+(deftest test-client-proxy-support
+  (with-http-ssl-servers basic-handler {}
+    (let [proxy-request (d/deferred)
+          proxy-server (tcp/start-server
+                        (fn [stream _]
+                          (d/future
+                            (try
+                              (let [stream-ch (:aleph/channel (meta stream))
```

Comment:
Doesn't have to be fixed here, but I always hated this. IIRC, sticking channels in metadata is the sole reason Manifold supports metadata abuse like this.

Reply:
Ah yes, see also our conversation about this not too long ago [in the Clojurians Slack](https://clojurians.slack.com/archives/C02H9GF74CF/p1731042930920799?thread_ts=1730280543.361439&cid=C02H9GF74CF)